ecosystem/sep-0030.md

```diff
@@ -257,7 +258,10 @@ Name | Type | Description
 `identities` | array | A list of identities that if any one is authenticated will be able to gain control of this account.
 `identities[].role` | string | Same as request. See above.
 `identities[].authenticated` | boolean | Optional. Present and true if the currently authenticated client is authenticated with the identity.
-`signer` | string | The signer public key that the server will sign transactions with for this account when requested by an authenticated user.
+`signer` | string | The most recently added signer public key that the server will sign transactions with for this account when requested by an authenticated user.
+`signers` | array | The signers' public keys that the server can sign transactions with for this account when requested by an authenticated user, ordered from most recently added to least recently added.
+`signers[].signer` | array | The signer public key that the server can sign transactions with for this account when requested by an authenticated user.
+`signers[].generated_at` | array | The RFC3339/ISO8601 timestamp of when the signer was generated and became available for signing.
```

Comment:
>How do we plan on maintaining the list of signers with the associated timestamp?

@howardtw This is out-of-scope for this document but the reference implementation will likely have a table of signers referencing an account, and as such in that relational model each account can have multiple signers. The associated timestamp would be stored in the row for the signer. The reference implementation currently shares a single key provided by an environment variable and so it might omit the timestamp from the API response.

Thinking about your question has made me question if we really need a timestamp. It's informative when debugging but a system is unlikely to use the value. The only decision a client needs to make is to determine which is the latest added signer and that can be done by looking at the top-level `signer` field or taking the first element of `signers`.

I'm learning towards omitting the date. We can always add the date later if it is needed.

@accordeiro @howardtw What do you think?

```suggestion
```

Reply:
> the reference implementation will likely have a table of signers referencing an account

This is out of scope. We can move the discussion to somewhere else. But I wonder how the recovery signer will know which signer is used by which account. It's up to the client to update the signer when a rotation happens. Unless we query each registered account, which would introduce further complication, we won't be able to know. Maybe it's not necessary to have such reference?

> Thinking about your question has made me question if we really need a timestamp.

The timestamp provides an insight for the client when the system updates the signing key. As a client of the recovery servers, it might be valuable to know when the system tends to rotate the signing key. Or even better, the reason that the system rotates the signing key. e.g. routinely semi-annual update, a potential threat was detected. I'm not sure whether revealing these information would impose security risks. For routinely update, maybe we can just document it in README. Just a thought from the client's perspective.

Reply:
I like keeping the `generated_at` time in there too. Having visibility into when important things happen like a key rotation can provide a better UX for users, and a better dev experience for client devs. Definitely a weak preference of mine though.

Reply:
👍 Sounds good, I'm convinced. I think I'm going to rename it to `added_at`. Does that sound smooth for you all?